rpi_backlight/cli.py

```diff
@@ -97,12 +97,19 @@ def main():
         return
 
     if args.set_power:
-        if any(
-            (args.get_brightness, args.set_brightness, args.get_power, args.duration)
-        ):
-            parser.error("-p/--set-power must be used without other options")
-        backlight.power = True if args.set_power == "on" else False
+        if any((args.get_brightness, args.set_brightness, args.get_power)):
+            parser.error("-p/--set-power may only be used with -d/--duration")
+        if args.set_power == "toggle":
+            with backlight.fade(duration=args.duration):
+                if backlight.brightness > 0:
+                    backlight.brightness = 0
+                else:
+                    backlight.brightness = 100
```

Comment:
Let's not assume that brightnes 0 means off and > 0 means on - that's probably not true on a Raspberry Pi, just Tinker Board :)

Instead, we might need something like this:

```suggestion
            if backlight.power:
                with backlight.fade(duration=args.duration):
                    backlight.brightness = 0
                backlight.power = False
            else:
                # Ensure brightness is 0 when we turn the display on
                backlight.brightness = 0
                backlight.power = True
                with backlight.fade(duration=args.duration):
                    backlight.brightness = 100
```

Reply:
In the above, this part is breaking the turning on slowly.
```
backlight.power = True
```

I had to take this line out, or else duration worked for turning off, but it would turn on instantly (would fade off, but instantly turn on)

Any suggestions?

Reply:
Also... I dont know if its a trick of my eye/brain, but it seems to fade off 2x as fast as it fades on.... When I provide -d 1, the off seems longer than the on.

Reply:
Had to set backlight.power = True AFTER the brightness change and not before

Reply:
This doesn't sound right :thinking:
Will have to do some testing.